### PR TITLE
SDIT-1836 Improve handling of merges for alerts

### PR DESCRIPTION
**Pull request by andymarke**
Now also tell DPS the list alerts that were retained. Also delete the orphaned mappings straight away rather than waiting for the associated domain event

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsSynchronisationService.kt [hunk 1]
```diff
@@ -187,16 +187,18 @@ class AlertsSynchronisationService(
     val offenderNo = prisonerMergeEvent.additionalInformation.nomsNumber
     val removedOffenderNo = prisonerMergeEvent.additionalInformation.removedNomsNumber
     val alerts = nomisApiService.getAlertsByBookingId(bookingId).alerts
-    val newAlerts = alerts.filter { mappingApiService.getOrNullByNomisId(bookingId = it.bookingId, alertSequence = it.alertSequence) == null }
+    val alertsMapping = alerts.groupBy { mappingApiService.getOrNullByNomisId(bookingId = it.bookingId, alertSequence = it.alertSequence) }
+    val newAlerts = alertsMapping[null] ?: emptyList()
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsSynchronisationService.kt:191]: divide alerts into those that have a mapping and those that don't

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsSynchronisationService.kt [hunk 2]
```diff
@@ -205,8 +207,10 @@ class AlertsSynchronisationService(
         mappingType = NOMIS_CREATED,
       )
     }
-
-    val mappingResponse = tryToCreateMappings(mappings, telemetry)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsSynchronisationService.kt:209]: delete mappings for the ones DPS told us were deleted

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsDpsApiService.kt [hunk 3]
```diff
@@ -13,6 +13,7 @@ import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.alerts.model.Merg
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.alerts.model.MigrateAlert
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.alerts.model.MigratedAlert
 import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.alerts.model.UpdateAlert
+import java.util.*
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsDpsApiService.kt:16]: star imports are back again, again.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsSynchronisationService.kt (hunk 2 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsSynchronisationService.kt:209]: Is there any value in keeping them so you have a history?

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsDpsApiService.kt (hunk 3 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsDpsApiService.kt:16]: I have no idea what has happened there an why ktlint has not complained - very odd

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsSynchronisationService.kt (hunk 2 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/alerts/AlertsSynchronisationService.kt:209]: Nah - it was just be a dangling record that will never be read

**Pull request merged**